include/dlaf/eigensolver/eigensolver/api.h

```diff
@@ -0,0 +1,32 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "dlaf/common/vector.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/types.h"
+
+namespace dlaf {
+namespace eigensolver {
+
+template <class T, Device device>
+struct ReturnEigensolverType {
```

Comment:
```suggestion
struct EigensolverReturnType {
```
?

Reply:
@rasolca @msimberg what about `EigensolverResultType`/`EigensolverResult`? any preference among all proposals?

Reply:
Weak preference:
1. `EigensolverResult`
2. `EigensolverResultType`
...
100. (<- is actually 100 in markdown but github renders it as a 3) `ReturnEigensolverType`